File: common/src/main/java/com/ssblur/scriptor/item/bound/BoundSword.java
package com.ssblur.scriptor.item.bound;

import com.ssblur.scriptor.color.CustomColors;
import dev.architectury.platform.Platform;
import dev.architectury.registry.client.rendering.ColorHandlerRegistry;
import net.fabricmc.api.EnvType;
import net.minecraft.world.item.SwordItem;
import net.minecraft.world.item.Tier;

public class BoundSword extends SwordItem {
  public BoundSword(Tier tier, Properties properties) {
    super(tier,properties);

    if(Platform.getEnv() == EnvType.CLIENT)
      ColorHandlerRegistry.registerItemColors((itemStack, t) -> t == 1 ? CustomColors.getColor(itemStack) : 0xFFFFFFFF, this);
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/item/tools/bound/BoundSword.kt
package com.ssblur.scriptor.item.tools.bound

import com.ssblur.scriptor.data.components.ScriptorDataComponents
import net.minecraft.ChatFormatting
import net.minecraft.network.chat.Component
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.SwordItem
import net.minecraft.world.item.Tier
import net.minecraft.world.item.TooltipFlag

class BoundSword(tier: Tier, properties: Properties): SwordItem(tier, properties) {
  override fun appendHoverText(
    itemStack: ItemStack,
    tooltipContext: TooltipContext,
    list: MutableList<Component>,
    tooltipFlag: TooltipFlag
  ) {
    super.appendHoverText(itemStack, tooltipContext, list, tooltipFlag)
    if(!itemStack.has(ScriptorDataComponents.EXPIRES))
      list.add(Component.translatable("lore.scriptor.bound_tool").withStyle(ChatFormatting.GRAY))
  }
}
